docs/sources/mimir/release-notes/v2.12.md

```diff
@@ -0,0 +1,119 @@
+- Responses with HTTP 4xx status codes are now treated as errors and used in `status_code` label of request duration metric.
+
+The default value of the following CLI flags has been changed:
+
+- `-blocks-storage.tsdb.head-postings-for-matchers-cache-max-bytes` from `10MB` to `100MB`.
+- `-blocks-storage.tsdb.block-postings-for-matchers-cache-max-bytes` from `10MB` to `100MB`.
+- `-blocks-storage.bucket-store.tenant-sync-concurrency` from `10` to `1`.
+- `-query-frontend.max-cache-freshness` from `1m` to `10m`.
+- `-distributor.write-requests-buffer-pooling-enabled` from `false` to `true`.
+- `-locks-storage.bucket-store.block-sync-concurrency` from `20` to `4`.
+- `-memberlist.stream-timeout` from `10s` to `2s`.
+- `-server.report-grpc-codes-in-instrumentation-label-enabled` from `false` to `true`.
+
+The following configuration options had been previously deprecated and are removed in Grafana Mimir 2.12:
+
+- The deprecated YAML setting `frontend.cache_unaligned_requests`
+
+The following configuration options are deprecated and will be removed in Grafana Mimir 2.14:
+
+- The CLI flag `-ingester.limit-inflight-requests-using-grpc-method-limiter`. It now defaults to `true`.
+- The CLI falg `-ingester.return-only-grpc-errors`. It now defaults to `true`. Note: in order to guarantee backwards compatibility when migrating from a version prior to `2.11`, it is necessary to first migrate to version `2.11`, and then to version `2.12`. Otherwise, it might happen that during the migration, some ingester pre-`2.11` errors with HTTP status code `4xx` will not be recognized as such, and the corresponding request will be repeated.
+- The CLI flag `-ingester.client.report-grpc-codes-in-instrumentation-label-enabled`. It now defaults to `true`.
+- The CLI flag `-distributor.limit-inflight-requests-using-grpc-method-limiter`. It now defaults to `true`.
+- The CLI flag `-distributor.enable-otlp-metadata-storage`. It now defaults to `true`.
+- The CLI flag `-querier.max-query-into-future`.
+-
+
+The following metrics are removed, deprecated or renamed:
+
+- `cortex_bucket_store_blocks_loaded_by_duration` has been removed.
+- `cortex_distributor_sample_delay_seconds` has been deprecated and will be removed in Mimir 2.14.
+- `prometheus_sd_failed_configs` has been renamed to `cortex_prometheus_sd_failed_configs`
+- `prometheus_sd_discovered_targets` has been renamed to `cortex_prometheus_sd_discovered_targets`
+- `prometheus_sd_received_updates_total` has been renamed to `cortex_prometheus_sd_received_updates_total`
+- `prometheus_sd_updates_delayed_total` has been renamed to `cortex_prometheus_sd_updates_delayed_total`
+- `prometheus_sd_updates_total` has been renamed to `cortex_prometheus_sd_updates_total`
+- `prometheus_sd_refresh_failures_total` has been renamed to `cortex_prometheus_sd_refresh_failures_total`
+- `prometheus_sd_refresh_duration_seconds` has been renamed to `cortex_prometheus_sd_refresh_duration_seconds`
```

Comment:
this refers to the prometheus service discovery while scraping. They should be irrelevant for us, right? In the name of brevity maybe we can omit them